sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -154,6 +171,20 @@ def _push_snapshots(self, snapshots: t.Iterable[Snapshot], overwrite: bool = Fal
             contains_json=True,
         )
 
+    @transactional()
```

Comment:
I think we only need to annotate the top-level `migrate` method, so that the entire migration happens as part of a single transaction and not just the version bump.

Reply:
Why has this been resolved? I see the decorator is still there.

Reply:
done